### DISCUSSION TITLE
A couple of questions (ID provider) and service tokens

### DISCUSSION THREAD

**Discussion by kotyara85** [ory/hydra#2954] [Q&A] [answered]
Hello everyone,
Can hydra do the next things:
1. Act as an ID provider to other OAUTH2 systems? Not public once, like a custom one
2. If so, is it possible to issue long live service account tokens based on a short lived token acquired by user in Hydra?
3. Machine to machine auth tokens?

Thanks!

**Comment by Benehiko** [comment, marked as answer]
Hi @kotyara85 

Thank you for your question :)

1. Please take a look at https://github.com/ory/hydra/discussions/2406
2. So you want Hydra to issue a long-lived token based on a short-lived session on the external ID provider? You can probably do something like that.
3. You can use the client credentials flow for that (https://www.ory.sh/hydra/docs/next/guides/oauth2-clients#client-credentials-flow)

**Reply by kotyara85** [reply]
@Benehiko thanks for the answers. 
regarding #2 were you referencing to check #1 answer?
Thank you!

**Reply by Benehiko** [reply]
yes, also please note that the solution would depend on your scenario as shown in the discussion. So first read through that and if you need more assistance, you can ask here :)

**Reply by vinckr** [reply]
Hey @kotyara85 ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝

**Reply by kotyara85** [reply]
Sorry for the later reply,
Actually this's what I was looking for - https://www.ory.sh/docs/hydra/guides/oauth2-grant-type-jwt-bearer
Thank you!
